pkg/shared/constants/event_signature_generator.go

```diff
@@ -0,0 +1,99 @@
+package constants
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/makerdao/vulcanizedb/pkg/eth"
+)
+
+func GetEventTopicZero(solidityEventSignature string) string {
+	eventSignature := []byte(solidityEventSignature)
+	hash := crypto.Keccak256Hash(eventSignature)
+	return hash.Hex()
+}
+
+func GetSolidityFunctionSignature(abi, name string) string {
+	parsedAbi, _ := eth.ParseAbi(abi)
+
+	if method, ok := parsedAbi.Methods[name]; ok {
+		return method.Sig
+	} else if event, ok := parsedAbi.Events[name]; ok {
+		return GetEventSignature(event)
+	}
+	panic("Error: could not get Solidity method signature for: " + name)
+}
+
+func GetEventSignature(event abi.Event) string {
+	types := make([]string, len(event.Inputs))
+	for i, input := range event.Inputs {
+		types[i] = input.Type.String()
+		i++
+	}
+
+	return fmt.Sprintf("%v(%v)", event.Name, strings.Join(types, ","))
+}
+
+type ContractMethod struct {
+	Name   string
+	Inputs []MethodInput
+}
+
+type MethodInput struct {
+	Type string
+}
+
+func GetOverloadedFunctionSignature(rawAbi, name string, paramTypes []string) string {
+	result, err := FindSignatureInAbi(rawAbi, name, paramTypes)
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+func FindSignatureInAbi(rawAbi, name string, paramTypes []string) (string, error) {
```

Comment:
There's a few public functions here that don't have tests. Are they tested in the transformers then?

Reply:
I think so - I believe most of these used to be private functions that were implementation details of tested public functions (e.g. `BiteSignature()` is tested, and that calls `GetEventTopicZero(biteMethod())` => `getSolidityFunctionSignature(Cat100ABI(), "Bite")` => `getContractABI("MCD_CAT_1.0.0")`.

That said in this _particular_ case there are tests for this function